src/V1/Payment.php

```diff
@@ -45,14 +45,33 @@ public function setCustomer(PaymentUserContract $user = NULL)
     }
     
 
-    public function pay($currency, $amount)
+    public function pay($currency, $amount, $metadata)
```

Comment:
```suggestion
    public function pay($currency, $amount, array $metadata = [])
```
If you introduce new param in the function, keep in mind that it must be backward compatible if possible.